src/metaopt/core/worker/experiment.py

```diff
@@ -154,10 +155,17 @@ def reserve_trial(self, score_handle=None):
         if not new_trials:
             return None
 
-        if score_handle is None:
-            selected_trial = random.sample(new_trials, 1)[0]
-        else:
-            raise NotImplementedError("scoring will be supported in the next iteration.")
+        if score_handle is not None and self.space:
```

Comment:
Based on trial_to_tuple code, only hyper-parameter values are passed to the score_handle, not the results? The score_handle need to keep internally results information to match hyper-parameters with them?

Reply:
I restrict that we should score a predictive performance scalar for trials which have not been evaluated yet. Algorithm will score based on what it has seen already from `observe`.

Reply:
score_handle could receive suspended or interrupted trials, which means there could be measurements available for the scoring functions. An example where this is necessary is FreezeThaw.

Reply:
#44